### PR TITLE
List Handles Access Rights

### PR DESCRIPTION
**Pull request by seanmcc-msft**

### CODE REVIEW

File: sdk/storage/Azure.Storage.Files.Shares/api/Azure.Storage.Files.Shares.net6.0.cs [hunk 1]
```diff
@@ -727,6 +728,14 @@ public partial class ShareFileHandle
         public string Path { get { throw null; } }
         public string SessionId { get { throw null; } }
     }
+    [System.FlagsAttribute]
+    public enum ShareFileHandleAccessRight
```
Comment (amnguye) [inline sdk/storage/Azure.Storage.Files.Shares/api/Azure.Storage.Files.Shares.net6.0.cs:732]: Should there be an 's' at the end so it's like `ShareFileHandleAccessRights`?

**Review by amnguye — APPROVED** [review]

File: sdk/storage/Azure.Storage.Files.Shares/src/autorest.md [hunk 2]
```diff
@@ -4,7 +4,7 @@ Run `dotnet build /t:GenerateCode` to generate code.
 
 ``` yaml
 input-file:
-    - https://raw.githubusercontent.com/Azure/azure-rest-api-specs/7dcd41cd28d46eb256bac034760a7e2f0a036238/specification/storage/data-plane/Microsoft.FileStorage/preview/2022-11-02/file.json
+    - C:\Users\seanmcc\git\azure-rest-api-specs\specification\storage\data-plane\Microsoft.FileStorage\preview\2023-01-03\file.json
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Files.Shares/src/autorest.md:7]: TODO fix this.

**Review by seanmcc-msft — COMMENTED** [review]
TODO manually test, make a recorded test if possible.

**Review by amnguye — APPROVED** [review]

File: sdk/storage/Azure.Storage.Files.Shares/src/autorest.md (hunk 2 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Files.Shares/src/autorest.md]: Fixed.

**Pull request merged**